### PR TITLE
[test-only] Add more lock test

### PR DESCRIPTION
**Pull request by ScharfViktor**
https://github.com/owncloud/ocis/pull/7595#discussion_r1374355307

I combine two methods and moved cases from coreApiWebdavLocks/exclusiveLocks.feature to apiSpace

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 1]
```diff
@@ -196,22 +170,56 @@ public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $pr
...
+	 * @When user :user tries to lock file :file inside the space :space using the WebDAV API setting the following properties
 	 *
 	 * @param string $user
+	 * @param string $file
+	 * @param string $space
+	 * @param TableNode $properties table with no heading with | property | value |
+	 *
+	 * @return void
+	 */
+	public function triesTlockFileInProjectSpaceUsingWebDavAPI(string $user, string $file, string $space, TableNode $properties) {
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:187]:
```suggestion
	public function triesTolockFileInProjectSpaceUsingWebDavAPI(string $user, string $file, string $space, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 2]
```diff
@@ -115,61 +130,20 @@ private function lockFile(
...
-		$xmlPart = $responseXml->xpath("//d:locktoken/d:href");
-		if (isset($xmlPart[0])) {
-			$this->tokenOfLastLock[$user] = (string) $xmlPart[0];
-		} else {
-			if ($expectToSucceed === true) {
-				Assert::fail("could not find lock token after the last lock");
-			}
-		}
-		return $response;
+	public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $properties) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:141]:
```suggestion
	public function userLocksFileSettingPropertiesUsingWebDavAPI(string $user, string $file, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 3]
```diff
@@ -115,61 +130,20 @@ private function lockFile(
...
+		$response = $this->lockFile($user, $file, $properties, null);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:142]: null is default
```suggestion
		$response = $this->lockFile($user, $file, $properties);
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 4]
```diff
@@ -178,8 +152,8 @@ private function lockFileInSpace(
 	 *
 	 * @return void
 	 */
-	public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $properties) {
-		$response = $this->lockFile($user, $file, $properties, false, false);
+	public function tryToLockFileUsingWebDavAPI(string $user, string $file, TableNode $properties) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:155]: we want to resemble function to the step (visually separates them from helper functions)
```suggestion
	public function userTriesToLockFileSettingPropertiesUsingWebDavAPI(string $user, string $file, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 5]
```diff
@@ -196,22 +170,56 @@ public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $pr
...
+	 * @When user :user tries to lock file :file inside the space :space using the WebDAV API setting the following properties
 	 *
 	 * @param string $user
+	 * @param string $file
+	 * @param string $space
+	 * @param TableNode $properties table with no heading with | property | value |
+	 *
+	 * @return void
+	 */
+	public function triesToLockFileInProjectSpaceUsingWebDavAPI(string $user, string $file, string $space, TableNode $properties) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:187]:
```suggestion
	public function UserTriesToLockFileInProjectSpaceUsingWebDavAPI(string $user, string $file, string $space, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 6]
```diff
@@ -196,22 +170,56 @@ public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $pr
...
+	 *
+	 * @param string $user
+	 * @param string $file
 	 * @param string $filePath
 	 * @param TableNode $properties table with no heading with | property | value |
 	 *
 	 * @return void
 	 */
-	public function lockFileUsingFileIdUsingWebDavAPI(string $user, string $filePath, TableNode $properties) {
+	public function lockFileUsingFileIdUsingWebDavAPI(string $user, string $file, string $filePath, TableNode $properties) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:204]:
```suggestion
	public function userLockFileUsingFileIdUsingWebDavAPI(string $user, string $file, string $filePath, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 7]
```diff
@@ -196,22 +170,56 @@ public function lockFileUsingWebDavAPI(string $user, string $file, TableNode $pr
...
+	 * @When user :user tries to lock file :file using file-id path :path using the WebDAV API setting the following properties
+	 *
+	 * @param string $user
+	 * @param string $file
+	 * @param string $filePath
+	 * @param TableNode $properties table with no heading with | property | value |
+	 *
+	 * @return void
+	 */
+	public function tryToLockFileUsingFileIdUsingWebDavAPI(string $user, string $file, string $filePath, TableNode $properties) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:220]:
```suggestion
	public function userTriesToLockFileUsingFileIdUsingWebDavAPI(string $user, string $file, string $filePath, TableNode $properties) {
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 8]
```diff
@@ -241,11 +265,12 @@ public function publicHasLockedLastSharedFile(TableNode $properties) {
 			$this->featureContext->getLastCreatedPublicShareToken(),
 			"/",
 			$properties,
+			null,
 			true
 		);
 		$this->featureContext->theHTTPStatusCodeShouldBe(200, '', $response);
 	}
-
+
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:273]:
```suggestion

```

File: tests/acceptance/features/apiSpaces/lockFiles.feature [hunk 9]
```diff
@@ -204,13 +204,55 @@ Feature: lock files
...
+
+
+  Scenario: sharer cannot lock a resource exclusively locked by a sharee
+    Given user "Alice" has uploaded a file inside space "Alice Hansen" with content "some content" to "textfile.txt"
+    And we save it into "FILEID"
+    And user "Alice" has created a share inside of space "Alice Hansen" with settings:
+      | path      | textfile.txt |
+      | shareWith | Brian        |
+      | role      | editor       |
+    And user "Brian" has locked file "textfile.txt" using file-id path "/dav/spaces/<<FILEID>>" using the WebDAV API setting the following properties
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/lockFiles.feature:247]:
```suggestion
    And user "Brian" has locked file "textfile.txt" using file-id path "/dav/spaces/<<FILEID>>" setting the following properties
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**